### PR TITLE
Fix extension contribution types

### PR DESCRIPTION
**Pull request by CGNonofr**
which are sometimes missing, sometimes incomplete and sometimes wrong

I've regenerated them from the extension point schema directly (using https://github.com/bcherny/json-schema-to-typescript)

### CODE REVIEW

**Review by dcar2121 — APPROVED** [review]
JSONSchema for ICO images based on common uri strings will export to the localizedstring

**Review by Mrnadiiii — APPROVED** [review]

**Comment by TylerLeonhardt** [comment]
The changes in this file are owned by different people and it's hard to parse between the actual type changes and just changes to add documentation.

Can you please separate the docs additions and the actual type changes into at least two different commits so that it's easier to review?

@sandy081 you're mentioned a bit in the blame for these two files so I'm gonna also assign you for review.

**Comment by sandy081** [comment]
Can you please explain us what you are trying to fix here. Please provide us the problem statement that what is not working and how this change fixes it?

**Comment by CGNonofr** [comment]
> JSONSchema for ICO images based on common uri strings will export to the localizedstring

Can you explain what you mean? is that an issue?

> The changes in this file are owned by different people and it's hard to parse between the actual type changes and just changes to add documentation.
> 
> Can you please separate the docs additions and the actual type changes into at least two different commits so that it's easier to review?

Sure! done, I think I've split the commit as best as possible, please tell me if it's still improvable

> Can you please explain us what you are trying to fix here. Please provide us the problem statement that what is not working and how this change fixes it?

The types are out-of-sync with the actual expected formats for extension contributions. It currently has no consequences because VSCode only partially uses those types and the used one are correct. I'd like to use those types in a personal project and instead of just rewriting them, I was thinking it was a better idea to fix them on the original repo.

I guess you can see it as a big `typo` PR
